plugins/woocommerce/src/Internal/Utilities/FilesystemUtil.php

```diff
@@ -121,4 +121,58 @@ protected static function initialize_wp_filesystem(): bool {
 
 		return is_null( $initialized ) ? false : $initialized;
 	}
+
+	/**
+	 * Validate that a file path is a valid upload path.
+	 *
+	 * @param string $path The path to validate.
+	 * @throws \Exception If the file path is not a valid upload path.
+	 */
+	public static function validate_upload_file_path( string $path ): void {
```

Comment:
These were moved into here to:

1. Improve testability of these methods in isolation
2. Allow reuse of validation and filters so we don't have multiple types of logic for different controllers should we ever need to validate other files